units/host_allocator.go

```diff
@@ -307,3 +286,33 @@ func (j *hostAllocatorJob) Run(ctx context.Context) {
 		"runner":                       scheduler.RunnerName,
 	})
 }
+
+func (j *hostAllocatorJob) setTargetAndTerminate(ctx context.Context, lenUpHosts int, hostQueueRatio float32, distro *distro.Distro) {
+	var killableHosts, newCapTarget int
+	if hostQueueRatio == 0 {
+		killableHosts = lenUpHosts
+		newCapTarget = 0
+	} else {
+		killableHosts = int(float32(lenUpHosts) * (1 - hostQueueRatio))
+		newCapTarget = lenUpHosts - killableHosts
+	}
+
+	if newCapTarget < distro.HostAllocatorSettings.MinimumHosts {
+		newCapTarget = distro.HostAllocatorSettings.MinimumHosts
+	}
+	// rough value to prevent killing hosts on low-volume distros
+	const lowCountFloor = 0
+	if killableHosts > lowCountFloor {
+
+		drawdownInfo := DrawdownInfo{
+			DistroID:     distro.Id,
+			NewCapTarget: newCapTarget,
+		}
+		err := amboy.EnqueueUniqueJob(ctx, j.env.RemoteQueue(), NewHostDrawdownJob(j.env, drawdownInfo, utility.RoundPartOfMinute(1).Format(TSFormat)))
+		if err != nil {
+			grip.Error(errors.Wrap(err, "Error drawing down hosts"))
```

Comment:
This should also use `message.WrapError` so that the error comes out nicely formatted. Otherwise, we'll just see a mostly unformatted error string in splunk with no other useful contextual information (like the drawdown info).